### PR TITLE
feat(detector-instana)!: change implementation to DetectorSync interface

### PR DESCRIPTION
**Pull request by david-luna**
## Context

Another PR moving away from the deprecated `Detector` interface in order to prepare the codebase for a fix for #2320. This time is for `Instana` detector.

My reasons to categorize it as a refactor are:
- SDK handles both interfaces using `detectResources` from `@opentelemetry/resources` package. [Ref](https://github.com/open-telemetry/opentelemetry-js/blob/54b14fbbe33e0cf38115ee8c5c934a4e27786186/packages/opentelemetry-resources/src/detect-resources.ts#L32)
- it is unlikely to be used oustide the SDK or auto-instrumentations

## Short description of the changes

- change `detect` method to be sync
- update tests

### Test changes

Previously the test where expecting the detector to throw but with the new implementation is not possible throw in a synchornous `detect` method when an async HTTP request fails. The behaviour has been changed to return an empty resource instead. This change is compatible with the current SDK since it actually does the same logic when a detector fails. You can check the `detectResources` implementation link above.

### CODE REVIEW

File: detectors/node/opentelemetry-resource-detector-instana/src/detectors/InstanaAgentDetector.ts
Comment (blumamir) [inline detectors/node/opentelemetry-resource-detector-instana/src/detectors/InstanaAgentDetector.ts:29]: While very unlikely, there might be users out there which can potentially break due to this change. 
Is there any reason to avoid marking it with `feat!` just in case?

File: detectors/node/opentelemetry-resource-detector-instana/test/InstanaAgentDetectorUnitTest.test.ts
Comment (blumamir) [inline detectors/node/opentelemetry-resource-detector-instana/test/InstanaAgentDetectorUnitTest.test.ts:108]:
```suggestion
    it('agent return empty resource if timeout', async () => {
```

**Review by blumamir — COMMENTED** [review]
LGTM, but I wonder if we should still mark it as breaking just in case.

**Comment by david-luna** [comment]
@basti1302 @kirrg001 is it okay for you to have this breaking change? If not we I can instead add a new detector implementing the sync interface

**Comment by basti1302** [comment]
Hi David, I do no longer work for IBM/Instana, so I don't have any particular opinion about this.

**Comment by david-luna** [comment]
Thanks @basti1302. Then we will go for the breaking change

**Review by blumamir — APPROVED** [review]

**Comment by blumamir** [comment]
As discussed in the last SIG and documented on the PR, it is very unlikely that people out there are calling the `InstanaAgentDetector` directly and will be affected by changing the `implements Detector` to `implements DetectorSync`.

Since the package is still experimental (major version 0), we can make this change as breaking.

@open-telemetry/javascript-approvers any objections before I merge this PR?

**Comment by pichlermarc** [comment]
> As discussed in the last SIG and documented on the PR, it is very unlikely that people out there are calling the `InstanaAgentDetector` directly and will be affected by changing the `implements Detector` to `implements DetectorSync`.
> 
> Since the package is still experimental (major version 0), we can make this change as breaking.
> 
> @open-telemetry/javascript-approvers any objections before I merge this PR?

I'm mainly waiting for someone from Instana to comment. Maybe we give it a couple more days until we merge this to ensure they have had enough time to comment.

**Comment by blumamir** [comment]
> > As discussed in the last SIG and documented on the PR, it is very unlikely that people out there are calling the `InstanaAgentDetector` directly and will be affected by changing the `implements Detector` to `implements DetectorSync`.
> > Since the package is still experimental (major version 0), we can make this change as breaking.
> > @open-telemetry/javascript-approvers any objections before I merge this PR?
> 
> I'm mainly waiting for someone from Instana to comment. Maybe we give it a couple more days until we merge this to ensure they have had enough time to comment.

 @kirrg001 can you please take a look?

**Comment by david-luna** [comment]
> > > As discussed in the last SIG and documented on the PR, it is very unlikely that people out there are calling the `InstanaAgentDetector` directly and will be affected by changing the `implements Detector` to `implements DetectorSync`.
> > > Since the package is still experimental (major version 0), we can make this change as breaking.
> > > @open-telemetry/javascript-approvers any objections before I merge this PR?
> > 
> > 
> > I'm mainly waiting for someone from Instana to comment. Maybe we give it a couple more days until we merge this to ensure they have had enough time to comment.
> 
> @kirrg001 can you please take a look?

@kirrg001 did you have the chance to look at this PR?

**Pull request merged**
